dbms/src/Functions/LeastGreatest.h

```diff
@@ -120,5 +121,363 @@ class FunctionVectorizedLeastGreatest : public IFunction
+        if (!const_columns.empty())
+        {
+            if (const_columns.size() == 1)
+            {
+                const_res = const_columns[0]->getValue<String>();
+            }
+            else
+            {
+                auto res = const_columns[0]->getValue<String>();
+                for (size_t i = 1; i < const_columns.size(); ++i)
+                    impl::constantConstant(const_columns[i]->getValue<String>(), res, collator, res);
+                if (string_columns.empty()) // fill result column
+                {
+                    block.getByPosition(result).column = block.getByPosition(result).type->createColumnConst(const_columns[0]->size(), Field(res));
+                    return;
+                }
+                else
+                {
+                    const_res = res;
+                }
+            }
+        }
+
+        // for string columns
+        std::vector<ColumnString::MutablePtr> results;
+        results.emplace_back(ColumnString::create());
+        results.emplace_back(ColumnString::create());
+
+        if (string_columns.size() == 1)
+            results[0] = ColumnString::create(*string_columns[0]);
+        else if (string_columns.size() >= 2)
+        {
+            for (size_t i = 1; i < string_columns.size(); ++i)
+            {
+                const DB::ColumnString * c0_string;
+                if (i == 1)
+                {
+                    c0_string = checkAndGetColumn<ColumnString>(string_columns[0]);
+                    auto size = c0_string->getOffsets().size();
+                    for (auto & res : results)
+                        res->getOffsets().resize(size);
+                }
+                else if (i % 2 == 0)
+                    c0_string = checkAndGetColumn<ColumnString>(results[1].get());
+                else
+                    c0_string = checkAndGetColumn<ColumnString>(results[0].get());
+                const auto * c1_string = string_columns[i];
+                impl::stringVectorStringVector(
+                    c0_string->getChars(),
+                    c0_string->getOffsets(),
+                    c1_string->getChars(),
+                    c1_string->getOffsets(),
+                    collator,
+                    results[i % 2]->getChars(),
+                    results[i % 2]->getOffsets());
+            }
+            if (const_columns.empty()) // no const columns, use string columns result
+            {
+                block.getByPosition(result).column = std::move(results[string_columns.size() % 2 == 0]);
+                return;
+            }
+        }
+        else // no string columns, use const columns result
+        {
+            auto col_str = ColumnString::create();
+            col_str->insert(Field(const_res));
+            auto const_res_col = ColumnConst::create(col_str->getPtr(), const_columns[0]->size());
+            block.getByPosition(result).column = std::move(const_res_col);
+            return;
+        }
```

Comment:
I think it's impossible to reach here.

Reply:
Yes, tidb will not push down this case, but I tested it in UT

Reply:
> Yes, tidb will not push down this case, but I tested it in UT

We can reach here only when `string_columns.size() == 0`. When `string_columns.size() == 0`, we can know that columns are all const columns, and the function has returned in line 402. So, it's impossible to hit this branch. By the way, which ut case could hit this branch?

Reply:
> > Yes, tidb will not push down this case, but I tested it in UT
> 
> We can reach here only when `string_columns.size() == 0`. When `string_columns.size() == 0`, we can know that columns are all const columns, and the function has returned in line 402. So, it's impossible to hit this branch. By the way, which ut case could hit this branch?

Ops, Removed